client/chat/channel-ban-user-dialog.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useCallback, useEffect, useRef } from 'react'
+import { ChannelModerationAction } from '../../common/chat'
+import { SbUser } from '../../common/users/sb-user'
+import { closeDialog } from '../dialogs/action-creators'
+import { CommonDialogProps } from '../dialogs/common-dialog-props'
+import { useForm } from '../forms/form-hook'
+import { TextButton } from '../material/button'
+import { Dialog } from '../material/dialog'
+import TextField from '../material/text-field'
+import { useAppDispatch } from '../redux-hooks'
+import { openSnackbar } from '../snackbars/action-creators'
+import { moderateUser } from './action-creators'
+
+interface BanUserModel {
+  banReason: string
+}
+
+type ChannelBanUserDialogProps = CommonDialogProps & {
+  channel: string
+  user: SbUser
+}
+
+export function ChannelBanUserDialog({
+  dialogRef,
+  onCancel,
+  channel,
+  user,
+}: ChannelBanUserDialogProps) {
+  const dispatch = useAppDispatch()
+  const inputRef = useRef<TextField>(null)
+
+  useEffect(() => {
```

Comment:
Instead of copy-pasting this another time, pull this into a hook that returns a ref to place on the thing that should be focused

Reply:
I needed this for something else, so this is now in `material/auto-focus.ts` and you can just use it here too